apps/api/src/domain/models/term/entities/term.entity.ts

```diff
@@ -78,8 +83,19 @@ export class Term extends Resource {
         return [];
     }
 
-    validateInvariants() {
-        return termValidator(this);
+    protected validateComplexInvariants(): InternalError[] {
+        const allErrors: InternalError[] = [];
+
+        const { term, termEnglish, id, published } = this;
+
+        if (!isStringWithNonzeroLength(term) && !isStringWithNonzeroLength(termEnglish))
+            allErrors.push(new TermHasNoTextInAnyLanguageError(id));
+
+        // TODO Validate inherited properties on the base class
```

Comment:
What about inherited `Complex` invariants?

Reply:
[Here](https://www.pivotaltracker.com/story/show/183014709) is a story to look at that.